### PR TITLE
etcd-backup-operator: add backup types and its auto generated files

### PR DESCRIPTION
**Pull request by fanminshi**
[skip ci]

### CODE REVIEW

File: etcd-backup-operator/pkg/util/k8sutil/k8sutil.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	return fmt.Sprintf("%s.%s.pod", podIP, p.Namespace)
+}
+
+// AddOwnerRefToObject appends the desired OwnerReference to the object
+func AddOwnerRefToObject(o metav1.Object, r metav1.OwnerReference) {
+	o.SetOwnerReferences(append(o.GetOwnerReferences(), r))
+}
+
+// AsOwner returns an owner reference set as the vault cluster CR
+func AsOwner(eb *api.EtcdBackup) metav1.OwnerReference {
```
Comment (hongchaodeng) [inline etcd-backup-operator/pkg/util/k8sutil/k8sutil.go]: MakeOwnerRef

File: etcd-backup-operator/pkg/util/k8sutil/crd.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	// (only required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+func CreateBackupCRD(clientset apiextensionsclient.Interface) error {
```
Comment (hongchaodeng) [inline etcd-backup-operator/pkg/util/k8sutil/crd.go]: We probably shouldn't manage CRD inside operator.

File: etcd-backup-operator/pkg/util/etcdutil/etcdutil.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package etcdutil
```
Comment (hongchaodeng) [inline etcd-backup-operator/pkg/util/etcdutil/etcdutil.go]: Let's reuse existing etcdutil/

File: etcd-backup-operator/example/backup.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+kind: "EtcdBackup"
+metadata:
+  name: example-etcd-cluster
+spec:
+  clusterName: example-etcd-cluster
+  storageType: s3
+  s3:
+    s3Bucket: jenkins-etcd-operator
+    prefix: prefix
+    awsSecret: aws
```
Comment (xiang90) [inline etcd-backup-operator/example/backup.yaml]: new line.

File: etcd-backup-operator/hack/push [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+: ${IMAGE:?"Need to set IMAGE, e.g. gcr.io/coreos-k8s-scale-testing/etcd-backup-operator:fanmin"}
+
+echo "building container..."
+docker build --tag "${IMAGE}" -f hack/Dockerfile . 1>/dev/null
+
+# For gcr users, do "gcloud docker -a" to have access.
+
+echo "pushing container..."
+docker push "${IMAGE}" 1>/dev/null
```
Comment (xiang90) [inline etcd-backup-operator/hack/push]: new line

**Comment by fanminshi** [comment]
@hongchaodeng made the pr only contains backup types and its auto-gen files

**Comment by fanminshi** [comment]
close in favor https://github.com/coreos/etcd-operator/pull/1451

**Pull request closed without merging**